src/NHibernate/Transaction/AdoNetWithSystemTransactionFactory.cs

```diff
@@ -298,6 +319,7 @@ public virtual void Prepare(PreparingEnlistment preparingEnlistment)
 								// this code runs from another thread and we cannot rely on Transaction.Current.
 								using (var tx = new TransactionScope(EnlistedTransaction))
 								{
+									_session.JoinTransaction();
```

Comment:
Required when both connection auto-enlistment and session auto-enlistment are disabled. Could have called directly `_connectionManager.EnlistIfRequired(System.Transactions.Transaction.Current)` instead, but well, simpler with above call.